cmd/network/verification.go

```diff
@@ -290,6 +291,36 @@ func (e *egressVerification) getSubnetId(ctx context.Context) (string, error) {
 		return e.cluster.AWS().SubnetIDs()[0], nil
 	}
 
+	if e.cluster.Hypershift().Enabled() {
+		instance, err := e.awsClient.DescribeInstances(ctx, &ec2.DescribeInstancesInput{
+			Filters: []types.Filter{
+				{
+					Name:   aws.String(fmt.Sprintf("tag:kubernetes.io/cluster/%s", e.clusterId)),
+					Values: []string{"owned"},
+				},
+				{
+					Name:   aws.String("tag:red-hat-managed"),
+					Values: []string{"true"},
+				},
+			},
+		})
+		if err != nil {
+			return "", fmt.Errorf("failed to find private subnets for %s: %w", e.clusterId, err)
+		}
+		if len(instance.Reservations) == 0 {
+			return "", fmt.Errorf("found 0 instances with kubernetes.io/cluster/%s=owned and red-hat-managed, consider the --subnet-id flag", e.clusterId)
+		}
+		if len(instance.Reservations[0].Instances) == 0 {
+			return "", fmt.Errorf("found 0 instances with kubernetes.io/cluster/%s=owned and red-hat-managed, consider the --subnet-id flag", e.clusterId)
+		}
+		if len(instance.Reservations[0].Instances[0].NetworkInterfaces) == 0 {
+			return "", fmt.Errorf("found 0 network interfaces of the worker node: %s, consider the --subnet-id flag", *instance.Reservations[0].Instances[0].InstanceId)
+		}
+
```

Comment:
should be a very rare case, but we might also check if there is more than one interface of the instance
like

```go

if len(instance.Reservations[0].Instances. NetworkInterfaces) > 1 {
			return "", fmt.Errorf("more than one interface found on the instance", e.clusterId)
		}
```